src/__tests__/postgres/integration.test.ts

```diff
@@ -394,4 +397,55 @@ describeMaybeSkip('integration (postgres)', () => {
     expect(errors).toBeUndefined()
     expect(data?.films.some((f: any) => f.sequel && f.sequel.id)).toBe(true)
   })
+
+  test('subscriptions', async () => {
+    const document = parse(`
+      subscription {
+        create 
+      }
+    `)
+
+    const sub = <AsyncIterator<ExecutionResult>>await subscribe(schema, document);
+    expect(sub.next).toBeDefined()
+    setImmediate(() => pubsub.publish('CREATE_ONE', { create: "FLUM!" }))
+
+    const { value: { errors, data } } = await sub.next()
+
+    expect(errors).toBeUndefined()
+    expect(data).toBeDefined()
+    expect(data.create).toBe('FLUM!')
+  })
+
+  // this test times out (sub.next() does not return)
```

Comment:
issue: `await sub.next()` translates to "pause executing this function until the Promise returned by `sub.next()` either resolves or rejects". That Promise will only resolve once something is finally published so execution is effectively paused indefinitely.

This is also why we have to use `setImmediate` in the test above. If we just called `pubsub.publish`, the publish would happen before we start awaiting `sub.next()` and we would also end up awaiting indefinitely. By using `setImmediate` we delay running `publish` long enough for `await sub.next()` to be ran first, which lets the Promise resolve when `publish` is finally called.

Reply:
I would expect that reversing the order of the publish/next lines would work just as well, but that doesn't seem to be the case; this timesout:
```
    const { value: { errors, data } } = await sub.next()
    pubSub.publish('CREATE_ONE', { create: "FLUM!" })
```
So why? Is it that sub.next() is doing some setup that isn't finished in time for the publish event? If so, then this should work:

```

    const { value: { errors, data } } = await sub.next()

    await new Promise(resolve => setTimeout(
      () => {
        pubSub.publish('CREATE_ONE', { create: "FLUM!" })
        resolve()
      }, 1000)
    )
```

But that doesn't work, either.

Reply:
Ok, now I see my dumb: I'm awaiting next(), so the next line isn't called.